### DISCUSSION TITLE
Ajuda com a tag <dialog>

### DISCUSSION THREAD

**Discussion by ihenrique** [frontendbr/forum#1610] [Discussão]
Alguém sabe como prevenir que o "ESC" feche o modal ? queria criar uma variação em meu projeto onde o <dialog> é um modal obrigatório e não pode ser fechado com ESC....

**Comment by Henriquepb22** [comment]
Fala Henrique, blz?
Só é adicionar o atributo `data-keyboard="false"` na tag do elemento modal que funciona!

**Comment by ihenrique** [comment]
> Fala Henrique, blz?
> Só é adicionar o atributo `data-keyboard="false"` na tag do elemento modal que funciona!

Não deu certo!

Se quiser dar uma olhada, deixei um exemplo no codepen da situação parecida com meu projeto atual.
https://codepen.io/ivhenrique/pen/LYYqLmj?&editable=true

**Comment by alexgwolff** [comment]
```javascript 
document.addEventListener('keydown', event => {
  if (event.key === 'Escape') {
    event.preventDefault();
  }
});
```




Não esqueça de remover esse evento quando não for mais utilizar.

**Comment by Henriquepb22** [comment]
> > Fala Henrique, blz?
> > Só é adicionar o atributo `data-keyboard="false"` na tag do elemento modal que funciona!
> 
> Não deu certo!
> 
> Se quiser dar uma olhada, deixei um exemplo no codepen da situação parecida com meu projeto atual.
> https://codepen.io/ivhenrique/pen/LYYqLmj?&editable=true

Eita, esqueci de perguntar, mas essa solução que eu dei é o modal do bootstrap, no caso vc tá usando puro em html/js né? so é fazer do jeito do @alexgwolff no seu js
